src/csharp/Tests/UnitTests/CaptureTests.cs

```diff
@@ -0,0 +1,508 @@
+                    Image image2_reference = image2.Reference();
+
+                    Assert.IsFalse(image1.NativeEquals(image2));
+                    Assert.AreNotSame(image1, image2);
+
+                    // Verify our images represent the same native object, but are not the same managed wrapper
+                    Assert.IsTrue(image1.NativeEquals(image1_reference));
+                    Assert.AreNotSame(image1, image1_reference);
+                    Assert.IsTrue(image2.NativeEquals(image2_reference));
+                    Assert.AreNotSame(image2, image2_reference);
+
+                    Assert.IsNull(c.Color);
+
+                    // Assign image1 to the capture, it is now owned by the capture object
+                    c.Color = image1;
+
+                    // both captures should refer to the same image
+                    Assert.IsTrue(image1.NativeEquals(c.Color));
+                    Assert.IsFalse(image2.NativeEquals(c.Color));
+                    Assert.IsTrue(image1.NativeEquals(captureReference.Color));
+                    Assert.IsFalse(image2.NativeEquals(captureReference.Color));
+
+                    // The reference should have its own wrapper that represents
+                    // the same native image
+                    Assert.IsTrue(c.Color.NativeEquals(captureReference.Color));
+                    Assert.AreNotSame(c.Color, captureReference.Color);
+
+                    c.Color = image2;
+
+                    // By re-assigning the image, the capture will dispose the original image
+                    _ = Assert.Throws(typeof(ObjectDisposedException), () =>
+                        {
+                            _ = image1.Size;
+                        });
+
+                    // Verify that the capture is now referring to image2.
+                    Assert.IsTrue(image2.NativeEquals(c.Color));
+                    Assert.AreSame(image2, c.Color);
+
+                    // Verify that the reference to the capture also is updated to the same new image
+                    Assert.IsFalse(image1_reference.NativeEquals(captureReference.Color));
+                    Assert.IsTrue(image2.NativeEquals(captureReference.Color));
+
+                    // The second capture will have its own wrapper
+                    Assert.AreNotSame(image2, captureReference.Color);
+
+                    // Assign the depth and IR properties to the same image
+                    c.Depth = image2;
+                    c.IR = image2;
+
+                    Assert.AreSame(image2, c.Depth);
+                    Assert.AreSame(image2, c.IR);
+                }
+            }
+        }
+
+        private System.WeakReference CreateWithWeakReference<T>(System.Func<T> factory)
+        {
+            return new System.WeakReference(factory());
+        }
+
+        // Helper function to implement basic open/close behavior
+        private void SetOpenCloseImplementation()
```

Comment:
Do you need this? Nothing is actually calling Device Open that I can see.

Reply:
I thought I needed it, but I guess I don't :)